src/libsystemd/sd-device/sd-device.c

```diff
@@ -2390,10 +2395,15 @@ static int device_cache_sysattr_value_full(sd_device *device, char *key, char *v
 }
 
 int device_cache_sysattr_value(sd_device *device, char *key, char *value, int error) {
-        return device_cache_sysattr_value_full(device, key, value, error, /* ignore_uevent = */ true);
+        char *value_stripped = strdup(value);
+        if (!value_stripped)
+                return -ENOMEM;
+
+        delete_trailing_chars(value_stripped, NEWLINE);
```

Comment:
Please move this in `device_cache_sysattr_value_full()`. Currently, `value_stripped` is leaked on failure.